src/string-calculator.ts

```diff
@@ -1,12 +1,235 @@
-export class StringCalculator{
+import { CommaFoundException } from "./helpers/comma-found-exception";
+import { CustomSeparatorFoundException } from "./helpers/custom-separator-found-exception";
+import { EndOfLineFoundException } from "./helpers/eol-found-exception";
+import { LastNumberMissedException } from "./helpers/last-number-exception";
+import { NegativeNumberException } from "./helpers/negative-number-exception";
+import { NumberNotNumericException } from "./helpers/number-not-numberic-exception";
 
-    public summation(numberList:string): string {
+enum ErrorCode {
+    CommaFoundErrorCode,
+    CustomSeparatorFoundErrorCode,
+    EndOfLineFoundErrorCode,
+    LastNumberMissedErrorCode,
+    NegativeNumberErrorCode,
+    NumberNotNumericErrorCode
+};
 
-        if(numberList.length=="".length){
+export class StringCalculator {
+
+    private numberList: string;
+    private customSep: string;
+    private errorMessage: string;
+    private lastErrorCode: ErrorCode = null;
+    private isUnexpectedEOLAdded: boolean = false;
+    private map = new Map<ErrorCode, any>([
```

Comment:
The problem with this map being static is that i call methods and parameters from this class whitch throws errors. The only solution i come up with is to make almost everything static or give as parameter to the function. I dont know if this is the solution.

Reply:
"or give as parameter to the function" do you see a flaw somewhere? I don't. Maybe if you clarify what you need it may evolve to a class that contains data for this error handling.